src/nupic/algorithms/KNNClassifier.py

```diff
@@ -777,7 +792,60 @@ def getPattern(self, idx, sparseBinaryForm=False, cat=None):
       else:
         pattern = nz
 
-      return pattern
+    return pattern
+
+
+  def getPartitionId(self, index):
+    """
+    Returns the partition Id associated with pattern idx.  Returns None
+    if no Id is associated with that
+    """
+    if (index < 0) or (index >= self._numPatterns):
+      raise RuntimeError("index out of bounds")
+    id = self._partitionIdList[index]
```

Comment:
avoid "id" as a variable name if possible since there is a Python builtin with that name